compiler-core/src/type_/expression.rs

```diff
@@ -611,6 +611,39 @@ impl<'a, 'b, 'c> ExprTyper<'a, 'b, 'c> {
                 .map_err(|e| convert_unify_error(e, value.type_defining_location()))?;
         }
 
+        if kind != AssignmentKind::Assert {
+            if let Pattern::Constructor {
+                constructor: PatternConstructor::Record { .. },
+                module: ref m,
+                type_: ref pattern_type,
+                ..
+            } = pattern
+            {
+                let pattern_type = Arc::clone(&pattern_type);
+                if let Type::App {
+                    name: type_name, ..
+                } = &*pattern_type
+                {
+                    // println!("pattern_type: {:?}", pattern_type);
+                    // println!("m: {:?}", m);
+                    // println!("type_name: {:?}", type_name);
+                    if let Ok(constructors) =
+                        self.environment.get_constructors_for_type(m, type_name)
+                    {
+                        // println!("constructors: {:?}", constructors);
+                        let constructors_n = constructors.len();
+                        if constructors_n != 1 {
+                            return Err(Error::IncorrectNumClausePatterns { // TODO_EXH_CHECK add and return a new kind of error here
```

Comment:
@lpil I chose this error as a placeholder for now but I'm curious, when is this one returned? (I didn't find any tests that mention it. Actually I didn't find any tests that check that errors get correctly returned...)

Reply:
Found it, it's for scenarios like this one:
```
case 1, 2 {
  1 -> Nil
}
```

Reply:
And I found a snapshot test at
`compiler-core/src/type_/tests/snapshots/gleam_core__type___tests__errors__wrong_number_of_subjects.snap`